packages/block-editor/src/components/image-size-control/index.js

```diff
@@ -33,6 +33,47 @@ export default function ImageSizeControl( {
 	const { currentHeight, currentWidth, updateDimension, updateDimensions } =
 		useDimensionHandler( height, width, imageHeight, imageWidth, onChange );
 
+	/**
+	 * Updates the dimensions for the given scale.
+	 * Handler for toggle group control change.
+	 *
+	 * @param {number} scale The scale to update the dimensions for.
+	 */
+	const handleUpdateDimensions = ( scale ) => {
+		/**
+		 * Check if scale is deselected from toggle group control option.
+		 */
+		if ( undefined === scale ) {
+			/**
+			 * Update dimensions to default.
+			 */
+			updateDimensions();
+			return;
+		}
+
+		/**
+		 * Calculate scaled width and height.
+		 */
+		const scaledWidth = Math.round( imageWidth * ( scale / 100 ) );
+		const scaledHeight = Math.round( imageHeight * ( scale / 100 ) );
+
+		/**
+		 * Update dimensions.
+		 */
```

Comment:
I'd say most of these comments are arbitrary since the code is self-explanatory.